workers.tf

```diff
@@ -23,21 +23,32 @@ resource "outscale_keypair" "workers" {
 
 resource "outscale_security_group" "worker" {
   description = "Kubernetes workers (${var.cluster_name})"
-  net_id      = outscale_net.net.net_id
+  net_id      = var.public_cloud ? null : outscale_net.net[0].net_id
 }
 
+resource "outscale_public_ip" "workers" {
+  count = var.public_cloud ? var.worker_count : 0
+}
+
+resource "outscale_public_ip_link" "workers" {
+  count     = var.public_cloud ? var.worker_count : 0
+  vm_id     = outscale_vm.workers[count.index].vm_id
+  public_ip = outscale_public_ip.workers[count.index].public_ip
+}
+
+
```

Comment:
In public cloud, workers will receive a publique IP even if you don't attach an EIP unless you specify `private_only` tag.
https://docs.outscale.com/en/userguide/About-Instances.html